benches/benches.rs

```diff
@@ -237,43 +352,105 @@ mod delayed_memory {
+            self.m.clear().await
+        }
+    }
 }
 
 fn bench_one_to_many(c: &mut Criterion) {
     #[cfg(feature = "redis-mem")]
-    let url = get_redis_url();
     let mut rng = CsRng::from_entropy();
 
-    #[cfg(feature = "redis-mem")]
+    #[cfg(any(feature = "redis-mem", feature = "postgres-mem"))]
     use delayed_memory::*;
 
     #[cfg(feature = "redis-mem")]
     bench_memory_one_to_many(
         "Redis",
         N_PTS,
-        async || DelayedMemory::new(RedisMemory::connect(&url).await.unwrap(), 1, 1),
+        async || DelayedMemory::new(RedisMemory::connect(&get_redis_url()).await.unwrap(), 1, 1),
         c,
-        DelayedMemory::clear,
+        DelayedMemory::<RedisMemory<_, _>>::clear,
         &mut rng,
     );
 
     #[cfg(feature = "redis-mem")]
     bench_memory_one_to_many(
         "Redis",
         N_PTS,
-        async || DelayedMemory::new(RedisMemory::connect(&url).await.unwrap(), 10, 1),
+        async || DelayedMemory::new(RedisMemory::connect(&get_redis_url()).await.unwrap(), 10, 1),
         c,
-        DelayedMemory::clear,
+        DelayedMemory::<RedisMemory<_, _>>::clear,
         &mut rng,
     );
 
     #[cfg(feature = "redis-mem")]
     bench_memory_one_to_many(
         "Redis",
         N_PTS,
-        async || DelayedMemory::new(RedisMemory::connect(&url).await.unwrap(), 10, 5),
+        async || DelayedMemory::new(RedisMemory::connect(&get_redis_url()).await.unwrap(), 10, 5),
+        c,
+        DelayedMemory::<RedisMemory<_, _>>::clear,
+        &mut rng,
+    );
+
+    #[cfg(feature = "postgres-mem")]
+    bench_memory_one_to_many(
+        "Postgres",
+        N_PTS,
+        async || {
+            let m = PostgresMemory::connect_and_init_table(
+                get_postgresql_url(),
+                "bench_memory_contention_m_1_var_1".to_string(),
+            )
+            .await
+            .unwrap();
+            DelayedMemory::new(m, 1, 1)
+        },
+        c,
+        DelayedMemory::<PostgresMemory<_, _>>::clear,
+        &mut rng,
+    );
+
+    #[cfg(feature = "postgres-mem")]
+    bench_memory_one_to_many(
+        "Postgres",
+        N_PTS,
+        async || {
+            let m = PostgresMemory::connect_and_init_table(
+                get_postgresql_url(),
+                "bench_memory_contention_m_10_var_1".to_string(),
+            )
+            .await
+            .unwrap();
+            DelayedMemory::new(m, 10, 1)
+        },
+        c,
+        DelayedMemory::<PostgresMemory<_, _>>::clear,
+        &mut rng,
+    );
+
+    #[cfg(feature = "postgres-mem")]
+    bench_memory_one_to_many(
+        "Postgres",
+        N_PTS,
+        async || {
+            let m = PostgresMemory::connect_and_init_table(
+                get_postgresql_url(),
+                "bench_memory_contention_m_10_var_5".to_string(),
+            )
+            .await
+            .unwrap();
+            DelayedMemory::new(m, 10, 5)
```

Comment:
We could consider factorizing all these instantiations in a loop over a set of delay parameters. This will put forward the fact that these delays are the same for all memories and will save many lines.